networkx/utils/decorators.py

```diff
@@ -225,3 +226,59 @@ def _open_file(func, *args, **kwargs):
         return result
 
     return _open_file
+
+
+def nodes_or_number(which_args):
+    """Decorator to allow number of nodes or container of nodes.
+
+    Parameters
+    ----------
+    which_args : int or sequence of ints
+        Location of the node arguments in args. Even if the argument is a
+        named positional argument (with a default value), you must specify its
+        index as a positional argument.
+        If more than one node argument is allowed, can be a list of locations.
+
+    Returns
+    -------
+    _nodes_or_numbers : function
+        Function which replaces int args with ranges.
+
+    Examples
+    --------
+    Decorate functions like this::
+
+       @nodes_or_number(0)
+       def empty_graph(nodes):
+           pass
+
+       @nodes_or_number([0,1])
+       def grid_2d_graph(m1, m2, periodic=False):
+           pass
+
+       @nodes_or_number(1)
+       def full_rary_tree(r, n)
+           # r is a number. n can be a number of a list of nodes
+           pass
+    """
+    @decorator
+    def _nodes_or_number(f, *args, **kw):
+        # form tuple of arg positions to be converted.
+        try:
+            iter_wa = iter(which_args)
+        except TypeError:
+            iter_wa = (which_args,)
+        # change each argument in turn
+        new_args = list(args)
+        for i in iter_wa:
+            n = args[i]
+            try:
+                nodes = list(range(n))
+                if n < 0:
+                    msg = "Negative number of nodes not valid:%i"%n
```

Comment:
My attitude is that `format` was added to Python by people smarter than me in order to fix some problem, so I should use the newest available feature whether I know the problem or not. Also, I trust methods more than statements, since I find them more predictable. These are not very good reasons, but they are my rules of thumb. 

Anyway, it's also valuable for string localization (i.e. translating strings to different languages), though that would really require using named format string templates (like `'{num_nodes} is not a valid number of nodes'.format(....)`).